ce/api/percentileanomaly.py

```diff
@@ -240,6 +240,34 @@ def canonical_timestamp(timestamp, timescale, timeidx):
                 projected_data[timescale][timestamp] = list(
                     np.percentile(anomalies, percentiles))
 
+        # converts a nested dictionary to a list storing data objects
+        def convert_to_li(dict):
```

Comment:
It's good that you set this functionality in a function!

I do think that you might be able to do it much more simply. You're on the right track with the `sort`ing using a function for the key. Did you know that your key function can return a tuple and Python with sort according to the first element of the tuple and then the next?

So you could combine a primary key as such: `lambda x: {"annual": 3, "seasonal": 2, "monthly": 1}[x]` with a secondary key that sorts on date. Then the whole function pretty much boils down to:

```python3
def convert_dict_to_sorted_list(d):
    ret_li = list(d)
    # define sort function
    ret_li.sort(key=sort_function)
    return ret_li
```

Reply:
> In particular, in the main loop starting on line 162, you already have the timeidx value direct from the rows retrieved from the CSV file

Ah, or use that! Even simpler.